src/lib/icon/icon.ts

```diff
@@ -317,24 +366,42 @@ export class MatIcon extends _MatIconMixinBase implements OnChanges, OnInit, Can
    * reference. This is required because WebKit browsers require references to be prefixed with
    * the current path, if the page has a `base` tag.
    */
-  private _prependCurrentPathToReferences(element: SVGElement) {
-    // @breaking-change 8.0.0 Remove this null check once `_location` parameter is required.
-    if (!this._location) {
-      return;
+  private _prependPathToReferences(path: string) {
+    const elements = this._elementsWithExternalReferences;
+
+    if (elements) {
+      elements.forEach((attrs, element) => {
+        attrs.forEach(attr => {
+          element.setAttribute(attr.name, `url('${path}#${attr.value}')`);
+        });
+      });
     }
+  }
 
+  /**
+   * Caches the children of an SVG element that have `url()`
+   * references that we need to prefix with the current path.
+   */
+  private _cacheChildrenWithExternalReferences(element: SVGElement) {
     const elementsWithFuncIri = element.querySelectorAll(funcIriAttributeSelector);
-    const path = this._location.pathname ? this._location.pathname.split('#')[0] : '';
+    const elements = this._elementsWithExternalReferences =
```

Comment:
Why not just initialize `_elementsWithExternalReferences` as an empty `Map`?

Reply:
I did it so we can check for its presence in `ngAfterViewChecked`, rather than having to check the size. I'm not sure whether I'm not micro-optimizing it though.